packages/model-viewer/src/features/scene-graph/nodes/primitive-node.ts

```diff
@@ -47,13 +48,23 @@ export class PrimitiveNode extends Node {
   private[$materials] = new Map<number, Material>();
   // Maps variant name to material index.
   private[$variantInfo]: Map<string, {material: Material, index: number}>;
-  // List of child nodes.
+  private[$defaultMaterialIdx]: number;
+
   constructor(
       mesh: Mesh, mvMaterials: Material[],
       correlatedSceneGraph: CorrelatedSceneGraph) {
     super(mesh.name);
     this[$mesh] = mesh;
     const {gltf, threeGLTF} = correlatedSceneGraph;
+    // Captures the primitive's default material.
+    const materialRef = correlatedSceneGraph.threeObjectMap.get(
+        mesh.material as MeshStandardMaterial);
+    if (materialRef != null) {
+      this[$defaultMaterialIdx] = materialRef.index;
+    } else {
+      console.error(
+          `Primitive (${mesh.name}) missing default material reference.`);
```

Comment:
Is this possible? If no material was  specified, we should have created a default material and then reference that, right? If it's just an assert, that's fine, I'm just curious.

Reply:
Correct the default material works and I don't think it would be possible for this to be null, at least not with todays codebase.